src/osd/OSD.cc

```diff
@@ -2205,7 +2205,18 @@ bool OSD::asok_command(string admin_command, cmdmap_t& cmdmap, string format,
       pg->unlock();
     }
     f->close_section();
-  } else {
+  } else if (admin_command == "compact") {
+    dout(1) << "triggering manual compaction" << dendl;
+    utime_t start = ceph_clock_now();
+    store->compact();
```

Comment:
Yeah that looks good. Not sure if that results in a timeout on the application end, may be we need to add(if possible) saying you might observe timeouts, if we have a lot to compact on the backend.